src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseIntegrationBambooBitbucketJiraTest.java

```diff
@@ -1060,4 +1058,38 @@ void testGetTemplateFilesWithContentShouldRedirect() throws Exception {
             return localRepository;
         });
     }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testGetParticipationFilesWithContentAtCommitShouldRedirect() throws Exception {
+        programmingExerciseIntegrationTestService.testRedirectGetParticipationRepositoryFilesWithContentAtCommit((exercise, files) -> {
+            LocalRepository localRepository = new LocalRepository("main");
+            var studentLogin = TEST_PREFIX + "student1";
+            try {
+                return hestiaUtilTestService.setupSubmission(files, exercise, localRepository, studentLogin);
+            }
+            catch (Exception e) {
+                fail("Test setup failed");
+            }
+            // we should never reach this but the code doesn't compile without it
+            return null;
+        });
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "editor1", roles = "EDITOR")
+    void testGetParticipationFilesWithContentAtCommitEditorForbidden() throws Exception {
+        programmingExerciseIntegrationTestService.testRedirectGetParticipationRepositoryFilesWithContentAtCommitForbidden((exercise, files) -> {
+            LocalRepository localRepository = new LocalRepository("main");
+            var studentLogin = TEST_PREFIX + "student1";
+            try {
+                return hestiaUtilTestService.setupSubmission(files, exercise, localRepository, studentLogin);
+            }
+            catch (Exception e) {
+                fail("Test setup failed");
+            }
```

Comment:
```suggestion
            return hestiaUtilTestService.setupSubmission(files, exercise, localRepository, studentLogin);
```
Exceptions make the test fail automatically, right?

Reply:
again, won't compile.